ddtrace/tracer.py

```diff
@@ -890,6 +891,60 @@ def flush(self):
         """Flush the buffer of the trace writer. This does nothing if an unbuffered trace writer is used."""
         self._writer.flush_queue()
 
+    def wrap_generator(
+        self,
+        name=None,  # type: Optional[str]
+        service=None,  # type: Optional[str]
+        resource=None,  # type: Optional[str]
+        span_type=None,  # type: Optional[str]
+    ):
+        def wrap_decorator(f):
+            # type: (AnyCallable) -> AnyCallable
+            # FIXME[matt] include the class name for methods.
+            span_name = name if name else "%s.%s" % (f.__module__, f.__name__)
+
+            # detect if the the given function is a coroutine to use the
+            # right decorator; this initial check ensures that the
+            # evaluation is done only once for each @tracer.wrap
+            if compat.iscoroutinefunction(f):
```

Comment:
@emmettbutler I'm actually not very familiar with coroutines, especially when they are working as generators. Do you know if this a use case we need within `wrap_generator`?